modules/command/dc/dc-challenge-check-command.js

```diff
@@ -0,0 +1,84 @@
+const Command = require('../command');
+const models = require('../../../models/index');
+
+/**
+ *  Challenges one DH
+ */
+class DCChallengeCommand extends Command {
+    constructor(ctx) {
+        super(ctx);
+        this.logger = ctx.logger;
+        this.transport = ctx.transport;
+        this.graphStorage = ctx.graphStorage;
+        this.challengeService = ctx.challengeService;
+    }
+
+    /**
+     * Executes command and produces one or more events
+     * @param command - Command object
+     * @param [transaction] - Optional database transaction
+     */
+    async execute(command, transaction) {
+        const {
+            dhId,
+            challengeId,
+            offerId,
+            dhIdentity,
+            litigationPrivateKey,
+        } = command.data;
+
+        const challenge = await models.challenges.findOne({ where: { id: challengeId } });
+
+        const replicatedData = await models.replicated_data.findOne({
+            where:
+                {
+                    offer_id: offerId, dh_id: dhId,
+                },
+        });
+
+        if (challenge.answer === challenge.expected_answer) {
```

Comment:
If challenge is not found in database this will be true.

Reply:
Challenge will be there :)